MedtrumKitUI/ViewController/MedtrumKitUICoordinator.swift

```diff
@@ -59,20 +67,97 @@ class MedtrumKitUICoordinator: UINavigationController, PumpManagerOnboarding, Co
+                self.pumpManager?.state.insulinType = insulinType
+                self.pumpManager?.notifyStateDidChange()
+                
+                self.navigateTo(.patchSettingsScreen)
+            }
+            return hostingController(rootView: InsulinTypeSelector(initialValue: allowedInsulinTypes[0], supportedInsulinTypes: allowedInsulinTypes, didConfirm: nextStep))
+            
+        case .patchSettingsScreen:
+            let nextStep = {
+                if let pumpManager = self.pumpManager {
+                    pumpManager.state.isOnboarded = true
+                    pumpManager.notifyStateDidChange()
+                }
+                
+                self.navigateTo(.pumpBaseSettingsScreen)
+            }
+            let viewModel = PatchSettingsViewModel(pumpManager)
+            return hostingController(rootView: PatchSettingsView(viewModel: viewModel, doDirtyCheck: false, nextStep: nextStep))
+            
+        case .deactivatePatchScreen:
+            let nextStep = { self.resetNavigationTo(.pumpBaseSettingsScreen) }
+            let viewModel = DeactivatePatchViewModel(pumpManager, nextStep)
+            return hostingController(rootView: PatchDeactivationView(viewModel: viewModel))
+            
+        case .pumpBaseSettingsScreen:
+            let viewModel = PumpBaseSettingsViewModel(pumpManager, {  self.navigateTo(.patchPrimingScreen) })
+            return hostingController(rootView: PumpBaseSettingsView(viewModel: viewModel))
+            
+        case .patchPrimingScreen:
+            let viewModel = PatchPrimingViewModel(pumpManager, { self.resetNavigationTo(.patchActivationScreen) })
+            return hostingController(rootView: PatchPrimingView(viewModel: viewModel))
+            
+        case .patchActivationScreen:
+            let nextStep = {
+                if let pumpManager = self.pumpManager, let pumpManagerOnboardingDelegate = self.pumpManagerOnboardingDelegate {
+                    pumpManagerOnboardingDelegate.pumpManagerOnboarding(didCreatePumpManager: pumpManager)
+                }
+                
+                self.resetNavigationTo(.settingsScreen)
+            }
+            let viewModel = PatchActivationViewModel(pumpManager, nextStep)
+            return hostingController(rootView: PatchActivationView(viewModel: viewModel))
+            
+        case .settingsScreen:
+            let toDeactivation = { self.navigateTo(.deactivatePatchScreen) }
+            let pumpRemoval = {
+                guard let completionDelegate = self.completionDelegate else {
+                    return
+                }
+                
+                completionDelegate.completionNotifyingDidComplete(self)
+            }
+            
+            let viewModel = MedtrumKitSettingsViewModel(self.pumpManager, toDeactivation, pumpRemoval)
+            return hostingController(rootView: MedtrumKitSettings(viewModel: viewModel, supportedInsulinTypes: allowedInsulinTypes))
         }
```

Comment:
Some things to keep into account for the activation process:
- The pump does timeout after 60 min (not 100% sure) if the user does not activate the patch after start of the activation process
- Communication interrupts will happen, you need to have a reliable way to handle that
- Also have a way to restore the activation process when the phone has been turned off, or for some other reason the app is closed